lib/API/JSON/v1/SendingTaskSubscribers.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace MailPoet\API\JSON\v1;
+
+use MailPoet\Listing;
+use MailPoet\Cron\CronHelper;
+use MailPoet\Models\Newsletter;
+use MailPoet\Models\ScheduledTask;
+use MailPoet\Config\AccessControl;
+use MailPoet\API\JSON\Error as APIError;
+use MailPoet\Settings\SettingsController;
+use MailPoet\WP\Functions as WPFunctions;
+use MailPoet\Models\ScheduledTaskSubscriber;
+use MailPoet\API\JSON\Endpoint as APIEndpoint;
+use MailPoet\Models\SendingQueue as SendingQueueModel;
+
+if (!defined('ABSPATH')) exit;
+
+class SendingTaskSubscribers extends APIEndpoint {
+  public $permissions = [
+    'global' => AccessControl::PERMISSION_MANAGE_EMAILS,
+  ];
+
+  /** @var Listing\Handler */
+  private $listing_handler;
+
+  /** @var SettingsController */
+  private $settings;
+
+  /** @var WPFunctions */
+  private $wp;
+
+  function __construct(
+    Listing\Handler $listing_handler,
+    SettingsController $settings,
+    WPFunctions $wp
+  ) {
+    $this->listing_handler = $listing_handler;
+    $this->settings = $settings;
+    $this->wp = $wp;
+  }
+
+  function listing($data = []) {
+    $newsletter_id = !empty($data['params']['id']) ? (int)$data['params']['id'] : false;
+    $tasks_ids = SendingQueueModel::select('task_id')
+      ->where('newsletter_id', $newsletter_id)
+      ->findArray();
+    if (empty($tasks_ids)) {
+      return $this->errorResponse([
+        APIError::NOT_FOUND => __( 'This email has not been sent yet.', 'mailpoet'),
+      ]);
+    }
+    $data['params']['task_ids'] = array_column($tasks_ids, 'task_id');
+    $listing_data = $this->listing_handler->get('\MailPoet\Models\ScheduledTaskSubscriber', $data);
```

Comment:
Shall we use some default order? I would think that failed subscribers should be in the beginning so that it is obvious what failed.

Reply:
This may be a good idea but it's not part of the specs.

Reply:
That is true, it is not part of the specs. But shouldn't we try to make the plugin the best for our users? Shouldn't we develop our features so that users would like to use them? We should go above and beyond to give our users a plugin they will love. Not just what is part of the specs.